### PR TITLE
bazel: migrate go:generate for staticcheck

### PR DESCRIPTION
**Pull request by Strum355**
Closes https://github.com/sourcegraph/sourcegraph/issues/54833

## Test plan

`bazel run //dev/linters/staticcheck:write_generated_bazelfile` and `cd dev/linters/staticcheck && go generate` as well as adding staticchecks to be ignored

### CODE REVIEW

**Comment by Strum355** [comment]
@jhchabran @burmudar Is there a catch-all target for generate that we should add this one to?

File: dev/linters/staticcheck/staticcheck.go [hunk 1]
```diff
@@ -1,4 +1,5 @@
-//go:generate go run ./cmd/gen.go BUILD.bazel
+//go:generate go run ./cmd/gen.go targets.bzl
+//go:generate buildifier -mode=fix targets.bzl analyzers.bzl
```
Comment (Strum355) [inline dev/linters/staticcheck/staticcheck.go:2]: Unfortunately dont have buildifier in buildkite, without it we have an extra newline :roll_eyes:

**Review by jamesmcnamara — APPROVED** [review]
I'll rubber stamp this in case it's blocked, but I can't purport to really get the ramifications of this yet.

File: dev/linters/staticcheck/staticcheck.go (hunk 1 above)
Comment (Strum355) [reply dev/linters/staticcheck/staticcheck.go]: Solved without buildifier

**Comment by burmudar** [comment]
> @jhchabran @burmudar Is there a catch-all target for generate that we should add this one to?

Yup. //dev:write_all_generated

**Pull request merged**